quixstreams/sinks/base/sink.py

```diff
@@ -17,6 +17,28 @@ class BaseSink(abc.ABC):
     Note that Sinks are currently in beta, and their design may change over time.
     """
 
+    def __init__(
+        self,
+        client_connect_cb: Optional[Callable[[Optional[Exception]], None]] = None,
+    ):
+        """
+        :param client_connect_cb: Am optional callback made once a client connection
+            is established. Callback expects an Exception or None as an argument.
```

Comment:
We could clarify this a bit, because it suggests that the cb will run _only_ `once a client connection is established`. We can get exceptions, maybe something along these lines:

```suggestion
        :param client_connect_cb: An optional callback invoked after attempting 
            to establish a client connection. The callback receives either a None for 
            successful connections or an Exception if the connection failed.
```

Reply:
Agreed, added some more details in the docstring :+1: